Advertisement/Android/src/main/java/com/applicaster/advertisement/Presenter.kt

```diff
@@ -0,0 +1,131 @@
+                override fun onAdLoaded() {
+                    super.onAdLoaded()
+                    adView.adLoaded(this@Presenter, null)
+                    adView.stateChanged(this@Presenter, AdViewState.Loaded)
+                }
+
+                override fun onAdFailedToLoad(errorCode: Int) {
+                    super.onAdFailedToLoad(errorCode)
+                    adView.adLoadFailed(this@Presenter, Exception(javaClass.simpleName
+                            + " could not load AdView. Failed with error code " + errorCode))
+                    adView.stateChanged(this@Presenter, AdViewState.Failed)
+                }
+
+                override fun onAdOpened() {
+                    super.onAdOpened()
+                    adView.stateChanged(this@Presenter, AdViewState.Impressed)
+                }
+
+                override fun onAdImpression() {
+                    super.onAdImpression()
+                    adView.stateChanged(this@Presenter, AdViewState.Impressed)
+                }
+
+                override fun onAdClosed() {
+                    super.onAdClosed()
+                    adView.stateChanged(this@Presenter, AdViewState.Closed)
+                }
+            }
+            publisherInterstitialAd.adUnitId = adConfig.adUnitId
+            loadInterstitialAd()
+        } else {
+            publisherAdView = PublisherAdView(context)
+
+            publisherAdView.adListener = object : AdListener() {
+                override fun onAdLoaded() {
+                    super.onAdLoaded()
+                    adView.adLoaded(this@Presenter, publisherAdView)
+                    adView.stateChanged(this@Presenter, AdViewState.Loaded)
+                }
+
+                override fun onAdFailedToLoad(errorCode: Int) {
+                    super.onAdFailedToLoad(errorCode)
+                    adView.adLoadFailed(this@Presenter, Exception(javaClass.simpleName
+                            + " could not load AdView. Failed with error code " + errorCode))
+                    adView.stateChanged(this@Presenter, AdViewState.Failed)
+                }
+
+                override fun onAdOpened() {
+                    super.onAdOpened()
+                    adView.stateChanged(this@Presenter, AdViewState.Impressed)
+                }
+
+                override fun onAdImpression() {
+                    super.onAdImpression()
+                    adView.stateChanged(this@Presenter, AdViewState.Impressed)
+                }
+
+                override fun onAdClosed() {
+                    super.onAdClosed()
+                    adView.stateChanged(this@Presenter, AdViewState.Closed)
+                }
+            }
+            //var v = View(context)
+            publisherAdView.adUnitId = adConfig.adUnitId
+            publisherAdView.setAdSizes(AdSizeMapper.map(adConfig.adType,adConfig.adSize))
+
+            loadBannerAd()
+        }
+    }
+    @SuppressLint("MissingPermission")
```

Comment:
I don't know if it is already on the documentation, but a comment of why we need to supress the warning will be very useful.